testing/test_script.py

```diff
@@ -1,18 +1,250 @@
+    file_conf = open("testing_data/model_config_auto.json", "r")
+    initial_config = json.load(file_conf)
+    file_conf.close()
+    file_conf = open("testing_data/model_config_auto.json", "w")
+    initial_config["command"] = "train"
+    initial_config["gpu"] = "[7]"
+    json.dump(initial_config, file_conf)
+
+
+def test_automate_training():
+    command = "ivadomed_automate_training -c testing_data/model_config_auto.json " \
+              "-p testing_data/hyperparameter_opt.json -n 1 --fixed-split"
+    subprocess.check_output(command, shell=True)
+
+    command = "ivadomed_automate_training -c testing_data/model_config_auto.json " \
+              "-p testing_data/hyperparameter_opt.json -n 1 --run-test --all-combin"
+    subprocess.check_output(command, shell=True)
+
+
+def test_create_json_3d_unet_test():
+    # modify train config
+    null = None
+    command = "cp ivadomed/config/config_tumorSeg.json testing_data/model_config_3d.json"
+    subprocess.check_output(command, shell=True)
+    file_conf = open("testing_data/model_config_3d.json", "r")
+    initial_config = json.load(file_conf)
+    file_conf.close()
+    file_conf = open("testing_data/model_config_3d.json", "w")
+    initial_config["command"] = "test"
+    initial_config["loader_parameters"] = {
+        "target_suffix": ["_lesion-manual"],
+        "roi_suffix": null,
+        "bids_path": "testing_data",
+        "roi_params": {
+            "suffix": null,
+            "slice_filter_roi": null
+        },
+        "contrast_params": {
+            "training_validation": ["T2w"],
+            "testing": ["T2w"],
+            "balance": {}
+        },
+        "slice_filter_params": {
+            "filter_empty_mask": False,
+            "filter_empty_input": True
+        },
+        "slice_axis": "sagittal",
+        "multichannel": False,
+        "soft_gt": False
+    }
+    initial_config["log_directory"] = "3d_test"
+    initial_config["UNet3D"] = {
+        "applied": True,
+        "length_3D": [48, 48, 16],
+        "stride_3D": [48, 48, 16],
+        "attention": False,
+        "n_filters": 8
+    }
+    initial_config["transformation"] = {
+        "Resample":
+            {
+                "wspace": 1,
+                "hspace": 1,
+                "dspace": 1,
+                "preprocessing": True
+            },
+        "CenterCrop": {"size": [48, 48, 16], "preprocessing": True},
+        "NumpyToTensor": {},
+        "NormalizeInstance": {"applied_to": ["im"]}
+    }
+    json.dump(initial_config, file_conf)
+
+
+def test_create_model_unet3d():
+    model = imed_models.UNet3D(in_channel=1, out_channel=1, n_filters=8)
+    torch.save(model, "model_unet_3d_test.pt")
+    os.makedirs("3d_test")
+    command = "cp model_unet_3d_test.pt 3d_test/best_model.pt"
+    subprocess.check_output(command, shell=True)
+
+
+def test_testing_unet3d():
+    subprocess.check_output(["ivadomed -c testing_data/model_config_3d.json"], shell=True)
```

Comment:
Linked to my previous comment but feels redundant with `test_training_time.py`

Reply:
`test_training_time.py` didn't actually called the training function so tht's why I went with that. Maybe `test_training_time.py` could be removed in the future